### PR TITLE
Add events for opening interval dropdowns

### PR DESCRIPTION
**Pull request by aneeshd16**
Related to p1732151233100969-slack-CFFF01Q4V

## Proposed Changes

Adds 2 new events to track when a interval dropdown has opened.
* `calypso_checkout_variant_dropdown_open`: On the checkout page.
* `calypso_plans_plan_type_selector_open`: Dropdown used on the plans step in onboarding flows and on the /plans page.

## Why are these changes being made?

* The data from these events will validate/invalidate the need for this experiment: p1732151233100969-slack-CFFF01Q4V

## Testing Instructions

* Go to `/start/plans`.
* Open the network inspector.
* Click on the interval dropdown. Confirm that you see a request for `http://pixel.wp.com/t.gif` with the following data:
image

* The event should be fired only once.
* Go to `/plans/<site slug>` and repeat the above steps.
* Go to checkout with any product in the cart and repeat the above steps. The event from checkout should have this data:
image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by oswian — COMMENTED** [review]
I can see the new events `calypso_plans_plan_type_selector_open` and `calypso_checkout_variant_dropdown_open `, however I noticed some of the other properties in the `http://pixel.wp.com/t.gif` call were different than the screenshots you included -- not sure if that is expected.

I tested using the live build: https://container-dazzling-bardeen.calypso.live

---

> Go to /start/plans ...

The `calypso_plans_plan_type_selector_open` value is present, however I don't see the `plans_intent` property that was in your screenshot.



> The event should be fired only once.

✅ 


> Go to /plans/<site slug> ...

Same as above, the `calypso_checkout_variant_dropdown_open ` value is present however there is no `plans_intent` property.




> Go to checkout with any product in the cart and repeat the above steps

The `calypso_checkout_variant_dropdown_open ` value is present, however instead of a `has_wpcom_plan` property I'm seeing `has_wpcom_plan_in_cart`.

File: client/my-sites/checkout/src/components/wp-order-review-line-items.tsx [hunk 1]
```diff
@@ -130,6 +131,10 @@ export function WPOrderReviewLineItems( {
 		);
 	}, [ responseCart.products ] );
 
+	const hasWPCOMPlanInCart = useMemo( () => {
```
Comment (sirbrillig) [inline client/my-sites/checkout/src/components/wp-order-review-line-items.tsx]: Minor, but I don't think this needs to be memoized. Boolean values passed as dependencies to `useCallback()` are compared by value not by reference, so even if this is recreated on each render, it won't accidentally trigger a recreation of the callback unless it changes.

Also, the overhead of calling `some` on the cart items on every render is relatively small, so I'm not sure this provides a lot there.

It's also fine if you want to keep it.

**Review by sirbrillig — COMMENTED** [review]
I didn't test this (I leave that to your team) but from a code perspective, this looks good to me.

File: client/my-sites/checkout/src/components/wp-order-review-line-items.tsx (hunk 1 above)
Comment (aneeshd16) [reply client/my-sites/checkout/src/components/wp-order-review-line-items.tsx]: Nice catch! I agree, `useMemo` is not necessary here.

**Comment by aneeshd16** [comment]
> I can see the new events calypso_plans_plan_type_selector_open and calypso_checkout_variant_dropdown_open , however I noticed some of the other properties in the http://pixel.wp.com/t.gif call were different than the screenshots you included -- not sure if that is expected.

Thanks for the review @oswian! I forgot to push after my latest commit and ended up posting the screenshots from my local machine. 😅 I have updated the PR, and the code should now match the screenshots!

**Pull request merged**